kitsune/wiki/views.py

```diff
@@ -1445,3 +1473,116 @@ def what_links_here(request, document_slug):
+        # This locale mapping is mentioned in kitsune/settings.py
+        for locale in header_locales:
+            if NON_SUPPORTED_LOCALES.get(locale) in translated_locales:
+                fallback_locale = NON_SUPPORTED_LOCALES[locale]
+                break
+        else:
+            # If above does not match, then check if we have custom Wiki Locale mapping
+            # for the user requested locale.
+            # This custom Wiki Locale Mapping is mentioned in kitsune/wiki/config.py
+            if request.LANGUAGE_CODE in FALLBACK_LOCALES:
+                for locale in FALLBACK_LOCALES[request.LANGUAGE_CODE]:
+                    if locale in translated_locales:
+                        fallback_locale = locale
+                        break
+                else:
+                    # If above does not match, then we would check if any locale of
+                    # Accepted_Language Header is in the Custom Wiki Locale mappping
+                    for locale in header_locales:
+                        if locale in FALLBACK_LOCALES:
+                            # Check if the document is trnaslated into the fallback locale
+                            for locale in FALLBACK_LOCALES[locale]:
+                                if locale in translated_locales:
+                                    fallback_locale = locale
+                                    break
+                            else:
+                                continue
+                            break
+                    else:
+                        # If all fails, return None as fallback Locale
+                        fallback_locale = None
+
+            # If Custom Wiki Locale Mapping is not available for the locale user requesed, then we
+            # will check if We have any fallback locale available in Custom Wiki locale
+            # mapping for the locales of Accpeted_Launguage Header.
+            else:
+                for locale in header_locales:
+                    if locale in FALLBACK_LOCALES:
+                        # Check if the document is trnaslated into the fallback locale
+                        for locale in FALLBACK_LOCALES[locale]:
+                            if locale in translated_locales:
+                                fallback_locale = locale
+                                break
+                        else:
+                            continue
+                        break
+                else:
+                    # If all fails, return None as fallback Locale
+                    fallback_locale = None
+    return fallback_locale
+
+
+# Import from django.utils.translation.trans_real and changed as below
+#  * Removed lower() from piece to get the locale name as needed
+#  * As we need only the locale name, added code to get only locale name
+def parse_accept_lang_header(lang_string):
+    """
+    Parses the lang_string, which is the body of an HTTP Accept-Language
```

Comment:
Entire section rewrite:
"Parse the lang_string from the body of the HTTP Accept-Language header
 Return a list of languages, ordered by their 'q' values
 Return an empty list if there are any format errors in lang_string"